cli/pack/ignore_test.go

```diff
@@ -0,0 +1,893 @@
+		result.dirOnly = true
+		return result
+	}
+	testCases := transformMapValues(ignorePatternBasicCases, toDirOnly)
+	checkIgnorePattern(t, testCases)
+}
+
+func Test_createIgnorePattern_trailingSpace(t *testing.T) {
+	toTrailingSpace := func(tc ignorePatternCase) ignorePatternCase {
+		result := tc
+		result.pattern = tc.pattern + strings.Repeat(" ", 1+len(tc.pattern)%3)
+		return result
+	}
+	testCases := transformMapValues(ignorePatternBasicCases, toTrailingSpace)
+	checkIgnorePattern(t, testCases)
+}
+
+// NOTE: For a new test that is not based on basic set the below snippet can be used.
+// func Test_createIgnorePattern_someNewTest(t *testing.T) {
+// 	testCases := ignorePatternCases{
+// 		"case_name1": {...},
+// 		"case_name2": {...},
+// 	}
+// 	checkIgnorePattern(t, testCases)
+// }
+
+type ignoreFilterCase struct {
+	// Ignore patterns.
+	patterns []string
+	// Files that are expected to be ignored/copied during copy.
+	// Every item here denotes file (not directory).
+	ignored []string
+	copied  []string
+}
+
+type ignoreFilterCases map[string]ignoreFilterCase
+
+// Check if all files are able to coexist within a single FS (the same path should not
+// refer to a file and a directory simultaneously).
+func validateIgnoreFilterCase(t *testing.T, tc ignoreFilterCase) {
+	files := slices.Concat(tc.ignored, tc.copied)
+	slices.Sort(files)
+	for i := 1; i < len(files); i++ {
+		assert.Falsef(t, strings.HasPrefix(files[i], files[i-1]+"/"),
+			"Invalid test case: %q and %q are not able to coexist within single FS",
+			files[i],
+			files[i-1],
+		)
+	}
+}
+
+func ignoreFilterCreateMockFS(t *testing.T, tc ignoreFilterCase) fs.FS {
+	validateIgnoreFilterCase(t, tc)
+
+	fsys := fstest.MapFS{}
+	if tc.patterns != nil {
+		fsys[ignoreFile] = &fstest.MapFile{
+			Data: []byte(strings.Join(tc.patterns, "\n")),
+			Mode: fs.FileMode(0644),
+		}
+	}
+	for _, name := range slices.Concat(tc.copied, tc.ignored) {
+		fsys[name] = &fstest.MapFile{
+			Mode: fs.FileMode(0644),
+		}
+	}
+	return fsys
+}
+
+func checkIgnoreFilter(t *testing.T, testCases ignoreFilterCases) {
+	basedst := t.TempDir()
+
+	// Do test
+	for name, tc := range testCases {
+		t.Run(name, func(t *testing.T) {
+			fsys := ignoreFilterCreateMockFS(t, tc)
+
+			filter, err := ignoreFilter(fsys, ignoreFile)
+			assert.Nil(t, err)
+			assert.NotNil(t, filter)
+
+			dst := filepath.Join(basedst, name)
+			err = os.MkdirAll(dst, 0755)
+			assert.Nil(t, err)
+
+			err = copy.Copy(".", dst, copy.Options{
+				FS: fsys,
+				Skip: func(srcinfo os.FileInfo, src, dest string) (bool, error) {
+					return filter(srcinfo, src), nil
+				},
+				PermissionControl: copy.AddPermission(0755),
+			})
+			assert.Nil(t, err)
+			for _, name := range tc.ignored {
+				assert.NoFileExists(t, path.Join(dst, name))
+			}
+			for _, name := range tc.copied {
+				assert.FileExists(t, path.Join(dst, name))
+			}
+		})
+	}
+}
+
+// Prepare basic test set for ignoreFilter function
```

Comment:
```suggestion
// Prepare basic test set for ignoreFilter function.
```
Here and in other comments, please add a point at the end.

Reply:
Fixed.